modules/message-lane/src/benchmarking.rs

```diff
@@ -200,6 +218,111 @@ benchmarks_instance! {
+	// * single relayer is rewarded for relaying single message;
+	// * relayer account does not exist (in practice it needs to exist in production environment).
+	//
+	// This is base benchmark for all other confirmations delivery benchmarks.
+	receive_delivery_proof_for_single_message {
+		let relayers_fund_id = relayer_fund_account_id::<T, I>();
+		let relayer_id: T::AccountId = account("relayer", 0, SEED);
+		let relayer_balance = T::account_balance(&relayer_id);
+		T::endow_account(&relayers_fund_id);
+
+		// send message that we're going to confirm
+		send_regular_message::<T, I>();
+
+		let proof = T::prepare_message_delivery_proof(MessageDeliveryProofParams {
+			lane: bench_lane_id(),
+			inbound_lane_data: InboundLaneData {
+				relayers: vec![(1, 1, relayer_id.clone())].into_iter().collect(),
+				latest_received_nonce: 1,
+				latest_confirmed_nonce: 0,
+			}
+		});
+	}: receive_messages_delivery_proof(RawOrigin::Signed(relayer_id.clone()), proof)
+	verify {
+		assert_eq!(
+			T::account_balance(&relayer_id),
+			relayer_balance + MESSAGE_FEE.into(),
+		);
+	}
+
+	// Benchmark `receive_messages_delivery_proof` extrinsic with fllowing conditions:
+	// * single relayer is rewarded for relaying two messages;
+	// * relayer account does not exist (in practice it needs to exist in production environment).
+	//
+	// Additional weight for paying single-message reward to the same relayer could be computed
+	// as `weight(receive_delivery_proof_for_two_messages_by_single_relayer)
+	//   - weight(receive_delivery_proof_for_single_message)`.
+	receive_delivery_proof_for_two_messages_by_single_relayer {
+		let relayers_fund_id = relayer_fund_account_id::<T, I>();
+		let relayer_id: T::AccountId = account("relayer", 0, SEED);
+		let relayer_balance = T::account_balance(&relayer_id);
+		T::endow_account(&relayers_fund_id);
+
+		// send message that we're going to confirm
+		send_regular_message::<T, I>();
+		send_regular_message::<T, I>();
+
+		let proof = T::prepare_message_delivery_proof(MessageDeliveryProofParams {
+			lane: bench_lane_id(),
+			inbound_lane_data: InboundLaneData {
+				relayers: vec![(1, 2, relayer_id.clone())].into_iter().collect(),
+				latest_received_nonce: 2,
+				latest_confirmed_nonce: 0,
+			}
+		});
+	}: receive_messages_delivery_proof(RawOrigin::Signed(relayer_id.clone()), proof)
+	verify {
+		assert_eq!(
+			T::account_balance(&relayer_id),
+			relayer_balance + (MESSAGE_FEE * 2).into(),
+		);
+	}
+
+	// Benchmark `receive_messages_delivery_proof` extrinsic with fllowing conditions:
+	// * two relayers are rewarded for relaying single message each;
+	// * relayer account does not exist (in practice it needs to exist in production environment).
+	//
+	// Additional weight for paying reward to the next relayer could be computed
+	// as `weight(receive_delivery_proof_for_two_messages_by_two_relayers)
+	//   - weight(receive_delivery_proof_for_two_messages_by_single_relayer)`.
```

Comment:
Is this because we'd have to do two separate transfers instead of one transfer with twice the funds? What other overhead costs are there for paying more relayers?

Reply:
Yes. Basically weight of this tx is now `base_weight + messages_count * message_overhead + relayers_count * relayer_overhead`. So there's one test to compute that `message_overhead` (which is ~ cost of db read) and another one for computing `relayer_overhead` (which is ~ cost of transfer).